rust-lib/src/lib.rs

```diff
@@ -346,3 +355,61 @@ pub extern "C" fn transfer(
     )
     .into()
 }
+
+fn _accept(
+    runtime: &COpaqueStruct,
+    consignment_bytes: *mut c_char,
+    reveal_outpoints: *mut c_char,
```

Comment:
How do you think, can we have a struct here instead of JSON serialization? This is a simple array of structs with three fields (txid, vout, blinding factor which is u64), must be easily interoperable with C... Probably this should be another PR though, since will require cbindgen for LNP/BP Core Library, but need not to forget

Reply:
we can easily get a `bp::blind::OutpointReveal` by parsing the single elements, but in this case `reveal_outpoints` is a `Vec<bp::blind::OutpointReveal>`, so, as for all other arrays that we parse in the ffi library, we need the JSON serialization. there should be another solution (I've investigated that in the past) but it seems hard to implement. so I'd open an issue to not forget this (to remove as much JSON serialization as possible), but for the moment I think it's best to stick with the easy solution

Reply:
I see. Still, doing data passing using JSON serialization... looks odd. Hard to believe that there is no more "normal" way of passing value structs from C to Rust...

I understand that we are here at the same level of not knowing how to do that properly, so let's leave it as is for now.

Reply:
issue #24 opened